File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        int n=nums.length;
        HashMap<Integer,Integer> hash= new HashMap<>();
        for( int i=0;i<=n-k;i++){
            HashSet<Integer> check= new HashSet<>();
            for( int j=i;j<i+k;j++){
                check.add(nums[j]);
            }
            for(int x: check){
                hash.put(x,hash.getOrDefault(x,0)+1);
            }
        }
        int ans=-1;
        for(Map.Entry<Integer, Integer> entry: hash.entrySet()){
            if( entry.getValue()==1){
                ans=Math.max(ans, entry.getKey());
            }
        }
        return ans;
    }
}

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.cpp
class Solution {
public:
    int largestInteger(vector<int>& arr, int k) {
        int left=0,right=0;
        int n=arr.size();
        unordered_map<int,int>hash;
        for(int i=0;i<=n-k;i++){
            unordered_set<int>seen;
            for(int j=i;j<i+k;j++){
                seen.insert(arr[j]);
            }
            for( int x: seen) hash[x]++;
        }
        int ans=-1;
        for(auto x: hash){
            if(x.second==1)ans=max(ans, x.first);
        }
        return ans;        
    }
};